matrix/MyMatrix.py

```diff
@@ -1,22 +1,114 @@
 class MyMatrix:
-    matrix = [[]]
-
-    def __init__(self, matrix):
+    def __init__(self, matrix, vector, results):
         self.matrix = matrix
+        self.size = len(matrix[0])
+        self.vector = vector
+        self.resultsGeneratedIncreateMatrix = results
+        self.results = [0 for i in range(0, self.size)]
+
+    # def __str__(self):
+    #     string = ""
+    #     for i in (0, self.size):
+    #        for j in (0, self.size):
+    #            string += (self.matrix[i][j])
+    #     return string
+    # to tak średnio chce działać, ale to działa jak toString() z javy
 
     def swapRows(self, x, y):
         temp = self.matrix[x]
         self.matrix[x] = self.matrix[y]
         self.matrix[y] = temp
 
     def swapColumns(self, x, y):
-        temp = []
+        temp = [None for i in range(0, self.size)]
         for i in range(len(self.matrix)):
-            temp.append(self.matrix[i][x])
+            temp[i] = self.matrix[i][x]
         for i in range(len(self.matrix)):
             self.matrix[i][x] = self.matrix[i][y]
         for i in range(len(self.matrix)):
             self.matrix[i][y] = temp[i]
 
-    def getValueFromPosition(self, x, y):
-       return self.matrix[x][y]
\ No newline at end of file
+    def gaussNone(self):
+        for i in range(0, self.size-1):
+            for j in range(0, self.size):
+                if self.matrix[i][i] == 0:
+                    raise Exception(ZeroDivisionError)
```

Comment:
Czemu wyjątek?
A to nie jest tak, że gdy jest zero w rogu to przekłądamy do pierwszego znalezionego niezera?

Reply:
"This version of Gaussian elimination is “‘naive” because it breaks down if a zero pivot entry is encountered at any step of the elimination, even though the system might be perfectly solvable without breakdown if the equations and/or unknowns had been given in a diﬀerent order" 
https://users.wpi.edu/~walker/MA2073/HANDOUTS/gaussian_elim.pdf
Basically tak ma być to tak zrobiłam

Reply:
Pod tym samym linkiem jest też wyjaśnione czemu size-1